### ISSUE TITLE
Undefined index: adminSortIndex

### ISSUE THREAD

**Issue by srenon** [Smile-SA/elasticsuite#562]
Exception #0 (Exception): Notice: Undefined index: adminSortIndex in /vendor/smile/elasticsuite/src/module-elasticsuite-catalog/Model/Layer/Filter/Attribute.php on line 279

### Preconditions

Magento Version : 2.1.9EE

ElasticSuite Version : 2.3.7

We are using a third-party extension to create products/attributes and it doesn't always add the 'adminSortIndex' for newly created attributes. To fix we normally have to resave the attributes option.

## Possible Solution 

            usort($items, function ($item1, $item2) {
               //Check if array key exist
                if(!array_key_exists('adminSortIndex', $item1) || !array_key_exists('adminSortIndex', $item2)){
                    return 0;
                }

                return $item1['adminSortIndex'] <= $item2['adminSortIndex'] ? -1 : 1;
            });

**Comment by afoucret** [comment]
A fix is available on branches 2.3.x, 2.4.x and master.

**Comment by srenon** [comment]
Thanks...

Are you going to do a new version release anytime soon?

**Comment by afoucret** [comment]
2.3.9 has been released and contains the fix.

BR,

**Comment by srenon** [comment]
@afoucret ... after upgrading to 2.3.9... I'm missing products from category

**Comment by srenon** [comment]
@afoucret ... the issue is not caused by this release but by this code https://github.com/Smile-SA/elasticsuite/pull/567

**Comment by DanielSousa** [comment]
@afoucret 
This issue still not fixed on 2.3.11
If you're using several languages and the translation value added in admin for some typo, or a bad copy paste has some space before or after the string the system will return false for `isset($items[$optionLabel])`
```php
            if ($optionLabel !== null && isset($items[$optionLabel])) {
                $items[$optionLabel]['adminSortIndex'] = $optionPosition;
                $items[$optionLabel]['value'] = $option['value'];
            }
```
Possible Solution:
```php
                        $optionLabel = trim($optionLabel);
                        if (isset($items[$optionLabel])) {
                            $items[$optionLabel]['adminSortIndex'] = $optionPosition;
                            $items[$optionLabel]['value']          = $option['value'];
                        }
```
Another improvement can be made:
the use of spaceship operator
```php
            usort($items, function ($item1, $item2) {
                return $item1['adminSortIndex'] <=> $item2['adminSortIndex'];
            });
```
